blog/source/_posts/MIDI-velocity-transformer.md

```diff
@@ -0,0 +1,266 @@
+   2. Start - Indicates the moment a key is pressed, measured in seconds.
+   3. End - Marks the second when the key is released.
+   4. Duration - calculated as the time elapsed between the key's press and release.
+   5. Velocity - ranging from 0 to 128, indicating the intensity of the key press.
+
+#### Quantization
+The way more suitable for quantization was what we engineered from this data.
+We used 4 values to describe notes:
+1. Pitch - same as above.
+2. Dstart - time elapsed after start of previous note.
+3. Duration - same as above,
+4. Velocity - same as above.
+
+This way we can achieve a more general data quantization, without having to consider
+variations in the song's tempo or using bins that cover the entire duration of 
+the sample. This approach ensures consistent quantization regardless of
+the length of the composition.
+
+We extracted 128-note samples which we quantized using 3 bins for dstart, 3 for duration and 3 for velocity.
+Pitch information remained the same.
+
+Here are **bin edges** we used:
+```
+dstart:
+  - 0.0
+  - 0.048177
+  - 0.5
+duration:
+  - 0.0
+  - 0.145833
+  - 0.450000
+velocity:
+  - 0.0
+  - 57.0
+  - 74.0
+  - 128.0
+```
+#### Quantization Samples
+<div style="-webkit-column-count: 2; -moz-column-count: 2; column-count: 2;">
+    
+  **Original** 
+        
+  **Quantized**
+
+</div>
+
+Franz Schubert / Franz Liszt: *Song Transcriptions: Aufenthalt, Gretchen am Spinnrade, Standchen von Shakespeare, Der Erlkonig*
+
+<div style="-webkit-column-count: 2; -moz-column-count: 2; column-count: 2;">
+
+  {% algrtmImg MIDI-velocity-transformer/samples/5002-real.png pianoroll 170px %}
+  {% algrtmAudio MIDI-velocity-transformer/samples/5002-real.mp3 %}
+        
+  {% algrtmImg MIDI-velocity-transformer/samples/5002-quantized.png pianoroll 170px %}
+  {% algrtmAudio MIDI-velocity-transformer/samples/5002-quantized.mp3 %}
+
+</div>
+
+
+
+Ludwig van Beethoven:  *Sonata No. 8 in C Minor, Op.13*
+<div style="-webkit-column-count: 2; -moz-column-count: 2; column-count: 2;">
+
+  {% algrtmImg MIDI-velocity-transformer/samples/9127-real.png pianoroll 170px %}
+  {% algrtmAudio MIDI-velocity-transformer/samples/9127-real.mp3 %}
+        
+  {% algrtmImg MIDI-velocity-transformer/samples/9127-quantized.png pianoroll 170px %}
+  {% algrtmAudio MIDI-velocity-transformer/samples/9127-quantized.mp3 %}
+
+</div>
+
+### Model Architecture
+{% algrtmImgBanner MIDI-velocity-transformer/transformer.png transformer%}
+A transformer built as described in [Attention is all you need](https://arxiv.org/abs/1706.03762) paper was used for this task.
+The important hyperparameters:
+| hyperparameter | number |
+| -------------- | :-----: |
+| Number of layers in encoder and decoder | **6** |
+| Nuber of heads in attention layers | **8** |
+| Dimension of encoder and decoder outputs | **512** |
+| Dimension of a hidden layer of position-wise fast-forward network from each layer of encoder and decoder | **2048** |
```

Comment:
Like I mentioned, we don't need to explain what transformers are. I think the only relevant information we should share about our architecture is the number of trainable parameters.

Also, I'm agains using diagrams from external sources, either make your own, or just link to the source :)